prow/config/presubmits.yaml

```diff
@@ -1245,6 +1245,55 @@ presubmits:
       - emptyDir: {}
         name: varlibdocker
 
+  - name: istio-push-images-2.3
+    trigger: (?m)^/test( | .* )push\-images,?($|\s.*)
+    rerun_command: /test push-images
+    skip_report: false
+    max_concurrency: 2
+    always_run: true
+    branches:
+    - ^maistra-2.3$
+    decorate: true
+    path_alias: istio.io/istio
+    labels:
+      preset-quay-pusher: "true"
+    spec:
+      containers:
+      - command:
+        - entrypoint
+        - sh
+        - -c
+        - HUB=quay.io/maistra TAG=PR-${PULL_NUMBER} make init push
```

Comment:
You can move these variables to under `env:`

Reply:
I can only move `HUB`, but not `TAG`, since it references the `PULL_NUMBER` env var. For symmetry, I added both here instead of adding one through `env`, and the other here.

Reply:
The make target could be `maistra-image.push`

Reply:
I think this is not going to work. I think prow will pass literal `${PULL_NUMBER}` and shell will not parse it.

In other code we handle this in a special fashion:

https://github.com/maistra/test-infra/blob/cc8b6734fd436395d6df50ca92cc5d61055f2e7a/prow/config/postsubmits.yaml#L590

https://github.com/maistra/test-infra/blob/cc8b6734fd436395d6df50ca92cc5d61055f2e7a/tools/automator.sh#L200

Reply:
I didn't want to change the Makefile in the repo, because then we have to make sure we add it every time we rebase, etc. I wanted to keep all the necessary changes here in test-infra.

Why wouldn't the shell parse it? IIUC, the `PULL_NUMBER` environment variable is available in the container.

Reply:
You can try. I hope you are right.

WRT to Makefile changes, we have our own, within `maistra` dir.